bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java

```diff
@@ -4408,7 +4408,7 @@ private static boolean isValidMapInsertion(BType mapType, BValue value) {
         }
```

Comment:
Not related to changes introduced in this PR, but we should check the constraint when the value is `null` right, whether it is a nil-able type?

The following doesn't fail atm, and allows an inherent type violating insertion!

```ballerina
import ballerina/io;

public function main(string... args) {
    map<string> m = { one: "one" };
    insertNilToMap(m);
    io:println(m);
}

function insertNilToMap(map<any> m) {
    m.two = ();
}
```

```cmd
$ ballerina run test.bal
{"one":"one", "two":()}
```

Reply:
Addressed with 2627ff2